### ISSUE TITLE
PDF browser plugin won´t work on 2.0.0?

### ISSUE THREAD

**Issue by smarquespt** [QupZilla/qupzilla#1913]
I´ve just started using release 2.0.0 for windows and pdf browser plugin won´t work on it. 

Was this option removed from QtWebengine or Qupzilla?

I´m using Qupzilla 2.0.0 (portable), Windows 7 and Pdf-XChange Viewer.

**Comment by Kizza00232Jera** [comment]
I have problem with flash (http://prntscr.com/an8xf2).. I've downloaded flash, and I have no problems on other browsers except this one (it wasn't problem on last versions). Would be nice if you fix it.

**Comment by nowrep** [comment]
You must use pepperflash, support for NPAPI plugins was removed.

**Comment by JHooverman** [comment]
@nowrep: Would you please add, which other plugins in addition to PDF are affected by the removal of NPAPI?

**Comment by nowrep** [comment]
@JHooverman No, I can't, because I don't know. There is a lot of Netscape plugins out there. But in most cases, they don't have PPAPI (pepper) version - so they are unsupported.

Flash, as the most used plugin, does have a pepper version, but that's an exception.

**Comment by smarquespt** [comment]
Oh, that´s really bad. I think @nowrep it´s correct. I don´t know any app thar has support for ppapi except flash. If there isn´t big problem opening pdf files in app, on the other hand I still need to use java  for tax purposes. Tax apps in Portugal still need to use java so for that reason qupzilla in unusable on that matter. But for the rest of the web that´s fine. 

@nowrep 

I guess ther´s no need to keep this issu open right?

**Comment by Martii** [comment]
> on the other hand I still need to use java for tax purposes.

https://blogs.oracle.com/java-platform-group/entry/moving_to_a_plugin_free *(deprecation notice of Java plugins)*

**Comment by smarquespt** [comment]
@Martii 

Hi MArti. Thanks for you link. But this info is made for developers. I´m not one of them. I just use their products as a regular user. But thanks anyway.

**Comment by Martii** [comment]
@smarquespt 
Here's a less techie link:

* https://hacked.com/oracle-retire-java-browser-plugin/

Point being your tax program shouldn't be using a Java *(JDK)* browser plugin too much longer. I have a similar issue arising with NPAPI and my security cams across my properties. I do however need to spend some time to see if there is a PPAPI plugin to NPAPI bridge... but that depends on if PPAPI goes away too.

**Comment by Martii** [comment]
Interesting... on ArchLinux it opened Okular, windowed within the browser, for a test pdf.

**Comment by Gamezertruth** [comment]
I have the same

**Comment by nowrep** [comment]
For Windows, Flash player must be installed in one of these locations:

```
C:\Windows\SysWOW64\Macromed\Flash\pepflashplayer*.dll
C:\Windows\System32\Macromed\Flash\pepflashplayer*.dll
```

**Comment by Thiudans** [comment]
I am having the same issue quth opening PDFs. I get a download window and I have to open the document manually from the Download directory. I am using openSUSE Tumbleweed and version 2.0.0 of Qupzilla.
